src/Text.Analyzers/Core (Net4)/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -0,0 +1,416 @@
+                                                                             DiagnosticSeverity.Warning,
+                                                                             isEnabledByDefault: false,
+                                                                             description: LocalizableDescription,
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/bb264492.aspx",
+                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+
+        private SemanticModel _model;
+        private NamingService _namingService;
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+            AssemblyRule,
+            NamespaceRule,
+            TypeRule,
+            MemberRule,
+            MemberParameterRule,
+            DelegateParameterRule,
+            TypeTypeParameterRule,
+            MethodTypeParameterRule,
+            AssemblyMoreMeaningfulNameRule,
+            NamespaceMoreMeaningfulNameRule,
+            TypeMoreMeaningfulNameRule,
+            MemberMoreMeaningfulNameRule,
+            MemberParameterMoreMeaningfulNameRule,
+            DelegateParameterMoreMeaningfulNameRule,
+            TypeTypeParameterMoreMeaningfulNameRule,
+            MethodTypeParameterMoreMeaningfulNameRule
+        );
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            // Do everything within a CompilationStartAction so we have access to the assembly name before the other actions run
+            analysisContext.RegisterCompilationStartAction(compilationStartContext =>
+            {
+                var assemblyName = compilationStartContext.Compilation.AssemblyName;
+
+                // Store the assembly name each time this analyzer runs so the appropriate custom dictionaries can be retrieved from the cache.
+                // The cache will remain in memory until Visual Studio is closed, so it's possible multiple solutions containing the same
+                // assembly names will be opened while the cache persists. This is not a problem because the cache associating assembly name
+                // to dictionaries is refreshed on the call to LoadDictionaries.
+                _namingService = new NamingService(assemblyName);
+                _namingService.LoadDictionaries(compilationStartContext.Options.AdditionalFiles, compilationStartContext.CancellationToken);
+
+                // CompilationStartAction doesn't support reporting diagnostics, so we have to check the assembly name in CompilationEndAction
+                compilationStartContext.RegisterCompilationEndAction(compilationEndContext =>
```

Comment:
:memo: `RegisterCompilationEndAction` doesn't do anything (dotnet/roslyn#11750), so this should be rewritten with an alternate approach that provides the desired behavior consistently.

Reply:
Dropped